apps/src/code-studio/components/progress/teacherPanel/StudentTable.jsx

```diff
@@ -42,29 +42,27 @@ class StudentTable extends React.Component {
     return isSelected ? [styles.tr, styles.selected] : styles.tr;
   };
 
-  render() {
-    const {
-      students,
-      onSelectUser,
-      selectedUserId,
-      levelsWithProgress,
-      isSortedByFamilyName,
-    } = this.props;
+  componentDidMount() {
```

Comment:
We were re-sorting on each render. Instead, we can just sort when the sorting style change or students change.

Reply:
Do we also need to re-sort when students change? ProgressTable does not and already has this functionality, but any new or modified student would be out of order until the sort is changed again.

Reply:
My main concern about this optimization would have been if perhaps the component is rendered both before and after initially fetching the list of students, but I see this component is only rendered in `TeacherPanel` after the student list is populated. I can't think of a reason the student list would be externally edited otherwise.